cmd/edgectl/aes_install.go

```diff
@@ -437,166 +429,100 @@ func (i *Installer) Perform(kcontext string) error {
-		case installedVersion != "":
-			i.ShowWrapped(fmt.Sprintf("-> Found an existing installation of Ambassador Edge Stack %s.", installedVersion))
-			i.show.Println()
-			i.ShowWrapped(abortExisting)
-			i.show.Println()
-			i.ShowWrapped(seeDocs)
-			i.Report("fail_existing_aes", ScoutMeta{"installing", i.version}, ScoutMeta{"found", installedVersion})
-			return errors.Errorf("existing AES %s found when installing AES %s", installedVersion, i.version)
-		default:
-			i.ShowWrapped(abortCRDs)
-			i.show.Println()
-			i.ShowWrapped(seeDocs)
-			i.Report("fail_existing_crds")
-			return errors.New("CRDs found")
-		}
+		i.log.Printf("when obtaining the chart manager: %s", err)
+		return err
 	}
 
-	if !alreadyApplied {
-		// Install the AES manifests
+	if err := chartRel.Sync(i.ctx); err != nil {
+		i.log.Printf("Failed to sync release: %s", err)
+		return err
+	}
 
-		i.ShowWrapped("Downloading images. (This may take a minute.)")
+	if !chartRel.IsInstalled() {
+		// First case: Ambassador is not installed at all: perform the installation
 
-		if err := i.ShowKubectl("install CRDs", crdManifests, "apply", "-f", "-"); err != nil {
-			i.Report("fail_install_crds")
-			return err
-		}
+		i.SetMetadatum("AES version being installed", "aes_version", downVersion)
+		i.version = downVersion
+		i.ShowWrapped(fmt.Sprintf("-> Installing the Ambassador Edge Stack %s.", i.version))
 
-		if err := i.ShowKubectl("wait for CRDs", "", "wait", "--for", "condition=established", "--timeout=90s", "crd", "-lproduct=aes"); err != nil {
-			i.Report("fail_wait_crds")
+		installedRelease, err := chartRel.InstallRelease(i.ctx)
+		if err != nil {
+			i.log.Printf("Installation of a new release failed: %s", err)
 			return err
 		}
 
-		if err := i.ShowKubectl("install AES", aesManifests, "apply", "-f", "-"); err != nil {
-			i.Report("fail_install_aes")
-			return err
-		}
+		i.log.Printf("New release installed successfully")
+		i.log.Printf("Config values: %+v", installedRelease.Config)
 
-		if err := i.ShowKubectl("wait for AES", "", "-n", "ambassador", "wait", "--for", "condition=available", "--timeout=90s", "deploy", "-lproduct=aes"); err != nil {
-			i.Report("fail_wait_aes")
-			return err
+		message := ""
+		if installedRelease.Info != nil {
+			message = installedRelease.Info.Notes
 		}
-	}
-
-	// Wait for Ambassador Pod; grab AES install ID
-	if err := i.loopUntil("AES pod startup", i.GrabAESInstallID, lc2); err != nil {
-		i.Report("fail_pod_timeout")
-		return err
-	}
+		i.ShowWrapped(message, "\n")
+		i.Report("deploy")
 
-	// Grab Helm information if present
-	if managedDeployment, err := i.CaptureKubectl("get deployment labels", "", "get", "-n", "ambassador", "deployments", "ambassador", "-Lapp.kubernetes.io/managed-by"); err == nil {
-		if strings.Contains(managedDeployment, "Helm") {
-			i.SetMetadatum("Cluster Info", "managed", "helm")
-		}
+	} else {
+		// Third case: Ambassador is currently installed. Don't do anything.
```

Comment:
What's the second case? 😂

Reply:
:-) There was a second case for a while... Fixing